extractor/src/test/java/com/example/extractor/MainTest.java

```diff
@@ -27,33 +27,15 @@ public void testParseConfigFromHeader() throws Exception {
         assertTrue(containsModule(xmlConfig, "AbbreviationAsWordInName"), "Config should contain module name 'AbbreviationAsWordInName'");
     }
 
-    private boolean containsModule(Configuration config, String moduleName) {
-        if (config.getName().equals(moduleName)) {
-            return true;
-        }
-
-        for (Configuration child : config.getChildren()) {
-            if (containsModule(child, moduleName)) {
-                return true;
-            }
-        }
-
-        return false;
-    }
-
     @Test
     public void testGeneratedConfigMatchesExpectedExample1() throws Exception {
         String exampleFilePath = "src/test/resources/com/puppycrawl/tools/checkstyle/checks/naming/abbreviationaswordinname/examples/Example1.java";
         String templateFilePath = "src/main/resources/config-template-treewalker.xml";
-        String outputFilePath = "src/test/resources/generated-config-example1.xml";
         String expectedFilePath = "src/test/resources/com/puppycrawl/tools/checkstyle/checks/naming/abbreviationaswordinname/configs/config-example1-expected.xml";
 
-        ConfigSerializer.serializeConfig(exampleFilePath, templateFilePath, outputFilePath);
-
-        String generatedContent = new String(Files.readAllBytes(Path.of(outputFilePath)));
+        String generatedContent = ConfigSerializer.serializeConfigToString(exampleFilePath, templateFilePath);
         String expectedContent = new String(Files.readAllBytes(Path.of(expectedFilePath)));
 
-        // Sort the properties in both the generated and expected content
```

Comment:
Please remove `ConfigSerializer.sortProperties(expectedContent)`
Serialization saves whole content, no separate properties.

Reply:
Instead of `assertEquals(sortedExpectedContent.trim()`

Please add to gradle Truth library from Google, and use assert from library

Reply:
> Please remove `ConfigSerializer.sortProperties(expectedContent)`
> Serialization saves whole content, no separate properties.

But, this method is needed to sort the expected files from Checkstyle's properties. If we don't the properties order don't match and test case fails. However, i have made changes such that generatedContent would be sorted and returned. Meaning, no explicit call for generatedContent.

Reply:
Should I refactor sorting to a separate class. But, to my understanding, we cant make test cases pass if we don;t sort.

Reply:
Sorting should be done before saving to file.

Expected file should have sorted properties.
No need to expect results of generation to be identical from what we have in examples.

Reply:
> Sorting should be done before saving to file.
> 
> Expected file should have sorted properties. No need to expect results of generation to be identical from what we have in examples.

i think there is confusion between expected and generated. we are currently not saving anything to disk. we are extracting from checkstyle exmaples which is expected. and, expected is in the order being parsed in the example.java files. And, while generating, we use configuration object, which doesn't preserve the order. And, when we call ConfigSerializer method, we sort the properties and present the generated conent which is sorted. And, then we explicityly sort the expected content before using assertTrue. I will try to push the recent patchset based on what i mentioned here. please correct me wherever you feel i am in sync with your thoughts.

Reply:
one time operation:
We should extract from ExampleX, sort properties, save to file and name it "-expected", store it to resources.
you and I will see such expected files and confirm that they are correctly generateed manually. There will not be a lot of them , around 10.

All next execution of tests should:
We should extract from ExampleX, sort properties, save to string. load content from related file "-expected", compare content.